compiler/src/main/java/io/neow3j/compiler/Compiler.java

```diff
@@ -803,23 +803,19 @@ public static NeoInstruction buildPushNumberInstruction(BigInteger number) {
         return new NeoInstruction(OpCode.get(insnBytes[0]), operand);
     }
 
-    /**
-     * Checks if the given class is an event.
-     *
-     * @param classDesc the descriptor of the class to check.
-     * @return true, if the given class is an event. False, otherwise.
-     */
-    public static boolean isEvent(String classDesc) {
-        String fqn = getFullyQualifiedNameForInternalName(
-                AsmHelper.stripObjectDescriptor(classDesc));
-        Class<?> clazz;
-        try {
-            clazz = Class.forName(fqn);
-        } catch (ClassNotFoundException e) {
+    public static boolean isEvent(String classDesc, CompilationUnit compUnit) {
+        char firstChar = classDesc.charAt(0);
```

Comment:
Is there any chance that the `classDesc` is `null` or `.charAt(0)` returns something absurd?

Reply:
Is there a better way to avoid unexpected exceptions?

Reply:
I think it is rather up to the caller of this method to make sure it doesn't pass null. I.e., if the caller passes null, then the caller has to be fixed, and this method should not just handle the exception and continue gracefully.